api/src/DataPersister/ContentNode/MaterialNodeDataPersister.php

```diff
@@ -19,10 +19,10 @@ public function __construct(
         );
     }
 
-    /**
-     * @param MaterialNode $materialNode
-     */
-    public function beforeCreate($materialNode): MaterialNode {
+    public function beforeCreate($data): MaterialNode {
+        /** @var MaterialNode $materialNode */
+        $materialNode = $data;
```

Comment:
Wieso musst du hier jetzt hier auf ein inline phpdoc definition zurückgreifen?

Reply:
Psalm error:
```
ERROR: ParamNameMismatch - src/DataPersister/ContentNode/MaterialNodeDataPersister.php:25:34 - Argument 1 of App\DataPersister\ContentNode\MaterialNodeDataPersister::beforeCreate has wrong name $materialNode, expecting $data as defined by App\DataPersister\Util\AbstractDataPersister::beforeCreate (see https://psalm.dev/230)
    public function beforeCreate($materialNode): MaterialNode {
```
Gibt 2 mögliche Lösungen:
- `$materialNode` im der ganzen Funktion umbenennen in `$data`
- `$data` zu `$matieralNode` assignen (code der Funktion ist dann etwas sprechender)

Für mich beides ok.